### PR TITLE
[bitnami/thanos] add dnsPolicy support in query deployment

### LINKED ISSUE #17517: [bitnami/thanos] Add support to template dnsPolicy
### Name and Version

bitnami/thanos 12.8.3

### What is the problem this feature will solve?

Thanos chart allows templating the value of dnsConfig in the deployment object, but does not allow to template the dnsPolicy field. If you need to set a specific DNS server to be used by thanos pods (useful when you use a SD based on DNS) you need to combine a dnsConfig that points to the DNS to be used + setting dnsPolicy to "None". Other charts like keycloak, redis, nginx-ingress, or rabbitmq already allow this. 

Reference of code where dnsConfig is used:
https://github.com/bitnami/charts/blob/main/bitnami/thanos/templates/query/deployment.yaml#L64

Some more info about dnsPolicy.
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

### What is the feature you are proposing to solve the problem?

Add support for templating the dnsConfig field, following the same approach used in other charts.

Example in rabbitmq:
https://github.com/bitnami/charts/blob/f559fa746836bf63e86ae1fe5ddbca4d72d14b68/bitnami/rabbitmq/templates/statefulset.yaml#L88

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by davidcarrera**
### Description of the change

This PR adds support to template the dnsPolicy field in the query deployment object. dnsPolicy controls the dns management policy in pods, allowing different policies to be selected. 

### Benefits

dnsPolicy allows adjusting of the dns configuration in a pod. So far only dnsConfig is possible in the chart, but extending this to dnsPolicy provides full control of the dns configuration.

### Possible drawbacks

No one envisioned.

### Applicable issues

- fixes #17517 

### Additional information

Reference: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution, could you please check and fix the failing pipelines? Thanks
* https://github.com/bitnami/charts/actions/runs/5486446277/jobs/9996553383?pr=17520
* https://github.com/bitnami/charts/actions/runs/5486446276/jobs/9996553192?pr=17520

**Comment by davidcarrera** [comment]
Fixed the issues, thanks

File: bitnami/thanos/Chart.yaml [hunk 1]
```diff
@@ -30,4 +30,4 @@ maintainers:
 name: thanos
 sources:
   - https://github.com/bitnami/charts/tree/main/bitnami/thanos
-version: 12.8.3
+version: 12.8.4
```
Comment (fmulero) [inline bitnami/thanos/Chart.yaml:33]: In these changes we are adding a new feature, support for `dnsPolicy`. According to [semver](http://semver.org) we should bump a minor version:

```suggestion
version: 12.9.0
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Great! thanks a lot David for your contribution.

I have only a little comment about your changes. Could you take it a glance?

File: bitnami/thanos/Chart.yaml (hunk 1 above)
Comment (davidcarrera) [reply bitnami/thanos/Chart.yaml]: I bumped chart version to 12.9.0 as indicated.

**Review by fmulero — APPROVED** [review]
Thanks @davidcarrera for your contribution. LGTM

**Pull request merged**